tat2xmpp/api/xmpp.go

```diff
@@ -19,31 +19,44 @@ func serverName(host string) string {
 }
 
 func getNewXMPPClient() (*xmpp.Client, error) {
-	xmpp.DefaultConfig = tls.Config{
-		ServerName:         serverName(viper.GetString("xmpp_server")),
-		InsecureSkipVerify: viper.GetBool("xmpp_insecure_skip_verify"),
-	}
+	hosts := strings.Split(viper.GetString("xmpp_server"), ",")
 
-	options := xmpp.Options{Host: viper.GetString("xmpp_server"),
-		User:          viper.GetString("xmpp_bot_jid"),
-		Password:      viper.GetString("xmpp_bot_password"),
-		NoTLS:         viper.GetBool("xmpp_notls"),
-		StartTLS:      viper.GetBool("xmpp_starttls"),
-		Debug:         viper.GetBool("xmpp_debug"),
-		Session:       viper.GetBool("xmpp_session"),
-		Status:        "",
-		StatusMessage: "",
-	}
+	var xmppClient *xmpp.Client
+	for index, host := range hosts {
+		xmppClient = nil
+		xmpp.DefaultConfig = tls.Config{
+			ServerName:         serverName(host),
+			InsecureSkipVerify: viper.GetBool("xmpp_insecure_skip_verify"),
+		}
 
-	xmppClient, err := options.NewClient()
+		options := xmpp.Options{
+			Host:          host,
+			User:          viper.GetString("xmpp_bot_jid"),
+			Password:      viper.GetString("xmpp_bot_password"),
+			NoTLS:         viper.GetBool("xmpp_notls"),
+			StartTLS:      viper.GetBool("xmpp_starttls"),
+			Debug:         viper.GetBool("xmpp_debug"),
+			Session:       viper.GetBool("xmpp_session"),
+			Status:        "",
+			StatusMessage: "",
+		}
 
-	if err != nil {
-		log.Panicf("getClient >> NewClient XMPP, err:%s", err)
-		return nil, err
+		var errNewClient error
+		xmppClient, errNewClient = options.NewClient()
+
+		if errNewClient != nil && index < len(hosts)-1 {
```

Comment:
This structure is cumbersome and decrease algorithm readability. Do you really need to claim the fail host and the next one in a single log message?

How about something like:

```go
var xmppClient *xmpp.Client
for _, host:= range hosts {
	log.Infof("Trying host %s", host)
	if xmppClient, err:= options.NewClient(); err != nil {
		log.Errorf("Host %s failed: %s", host, err)
	}
	break
}
if xmppClient == nil {
	log.Panicf("Everybody died. You are dead. Game over.")
}
```